### PR TITLE
feat: add CLI improvements and database merge tool

### PR DESCRIPTION
**Pull request by benroeder**
## Summary
- Add new CLI flags for flexible database and search configuration
- Add rclip-db-merge tool for merging multiple rclip databases
- Add comprehensive test coverage for all new features

### New CLI Flags
- **--db-path**: Specify custom database location instead of default platform directory
- **--search-dir**: Search in a specific directory instead of current working directory  
- **--index-only**: Index images without performing a search
- **--db-cache-size**: Configure SQLite cache size in MB for performance tuning

### New Database Merge Tool
- **rclip-db-merge**: Command-line tool to merge two rclip SQLite databases
- Intelligent conflict resolution (newer images win)
- Preserves deleted image status
- Progress tracking with tqdm
- Performance optimizations (batching, WAL mode, configurable memory)
- --check-versions flag for pre-merge validation
- --dry-run mode for previewing operations
- Database integrity checking
- Path normalization for consistent filepath comparison

### Testing
- Added 17 tests for rclip-db-merge tool
- Added 17 tests for new CLI flags
- All tests passing with 100% coverage of new features

### Documentation
- Updated README with Advanced Options section
- Documented all new CLI flags and their usage
- Added rclip-db-merge tool documentation

## Test Plan
- [x] Run full test suite: `pytest`
- [x] Test --db-path with custom locations
- [x] Test --search-dir with different directories
- [x] Test --index-only flag
- [x] Test --db-cache-size with various sizes
- [x] Test rclip-db-merge with sample databases
- [x] Verify all edge cases and error conditions

### CODE REVIEW

File: README.md
Comment (yurijmikhalevich) [inline README.md:1]: Hi! Thank you for your interest in contributing to rclip and for drafting this PR!

This PR introduces a lot of changes, which should've been a few separate PRs (with preliminary discussions):
- custom DB location should be a PR
- custom search dir needs more thought because it complicates rclip; searching in the current directory is rclip's feature, not a bug
- db index cache should be a separate PR
- merging databases should be a separate PR

File: README.md [hunk 1]
```diff
@@ -136,6 +136,119 @@ rclip -p kitty
...
+
+By default, rclip stores its database in your system's application data directory. You can specify a custom location:
+
+```bash
+# Use a custom database file
+rclip --db-path /path/to/my/photos.db "sunset"
+
+# Store database on an external drive
+rclip --db-path /mnt/photos/rclip.db "vacation"
+```
```
Comment (yurijmikhalevich) [inline README.md:151]: Can you please share a user story backing up the need for a custom database location?

File: README.md [hunk 2]
```diff
@@ -136,6 +136,119 @@ rclip -p kitty
...
+
+Search for images in a specific directory without changing your current directory:
+
+```bash
+# Search in a different directory
+rclip --search-dir /path/to/photos "cats"
+
+# Search external drive
+rclip --search-dir /mnt/backup/photos "birthday"
+```
```
Comment (yurijmikhalevich) [inline README.md:163]: Searching in the current directory is a feature of `rclip`, not a bug. It makes it easy and intuitive to use rclip, akin to `rgrep`.

I am wary of introducing another way to search to `rclip` because it makes rclip CLI more confusing.

Can you please share a user story that supports the need for a `--search-dir` CLI argument.

File: README.md [hunk 3]
```diff
@@ -136,6 +136,119 @@ rclip -p kitty
...
+```bash
+# Just index the current directory
+rclip --index-only
+
+# Index a specific directory
+rclip --index-only --search-dir /path/to/new/photos
+
+# Index to a custom database
+rclip --index-only --db-path /path/to/custom.db --search-dir /photos
+```
```
Comment (yurijmikhalevich) [inline README.md:178]: This is nice, but I am not sure if we should merge it as-is because how it's implemented confuses the rclip `--help` message.

Please see my comment [here](https://github.com/yurijmikhalevich/rclip/pull/190#discussion_r2280699745).

File: README.md [hunk 4]
```diff
@@ -136,6 +136,119 @@ rclip -p kitty
...
+
+```bash
+# Use 64MB cache for faster indexing (default is 2MB)
+rclip --index-only --db-cache-size 64
+
+# Use 128MB cache when indexing millions of images
+rclip --index-only --db-cache-size 128 --search-dir /massive/photo/library
+```
+
+Note: Increasing cache size primarily speeds up indexing operations. For searching, the default cache is usually sufficient unless you have millions of images in a single directory.
```
Comment (yurijmikhalevich) [inline README.md:192]: This is an interesting area of improvement. Thank you for proposing this!

I wonder:
1. What is the speedup that you saw from changing this?
2. Should we make 32Mb the default db cache for indexing and remove the argument?

File: README.md [hunk 5]
```diff
@@ -136,6 +136,119 @@ rclip -p kitty
...
+```
+
+### Database Management
+
+#### Merging Databases
+
+The `rclip-db-merge` tool allows you to combine multiple rclip databases into one. This is useful when:
+- You have indexed photos on different computers
+- You want to combine databases from different photo collections
+- You need to consolidate multiple partial indexes
```
Comment (yurijmikhalevich) [inline README.md:213]: How does it handle image paths? I guess that for cases 1 and 2 the paths of the indexed images would also change, correct?

File: tests/unit/test_merge_dbs.py [hunk 6]
```diff
@@ -0,0 +1,619 @@
...
+                filepath TEXT NOT NULL UNIQUE,
+                modified_at DATETIME NOT NULL,
+                size INTEGER NOT NULL,
+                vector BLOB NOT NULL,
+                indexing BOOLEAN
+            )
+        """)
+
+    con.execute("CREATE TABLE IF NOT EXISTS db_version (version INTEGER)")
+    con.execute("INSERT INTO db_version(version) VALUES (2)")
```
Comment (yurijmikhalevich) [inline tests/unit/test_merge_dbs.py:30]: If we are going to move forward with this feature, we should share these SQL queries with the logic that initializes the rclip database to prevent them from going out of sync.

**Review by yurijmikhalevich — CHANGES_REQUESTED** [review]
Thank you for proposing these changes! Please see my comments above and let me know if you have any thoughts.
